File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int n=nums.size();
        int l=0;
        int r=n-1;
        int start=-1,end=-1;
        int mid;
        while(l<=r){
            mid=l+(r-l)/2;
            if(nums[mid]>target){
                r=mid-1;
            }else if(nums[mid]==target){
                start=mid;
                r=mid-1;
            }else{
                l=mid+1;
            }
        }
        l=0;
        r=n-1;
        while(l<=r){
            mid=l+(r-l)/2;
            if(nums[mid]>target){
                r=mid-1;
            }else if(nums[mid]==target){
                end=mid;
                l=mid+1;
            }else{
                l=mid+1;
            }
        }
        return {start,end};
        
        
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
      
      def findfirst(nums, target):
        l, r = 0, len(nums) - 1
        first = -1
        while l <= r:
          middle = (l + r) // 2
          if nums[middle] == target:
              first = middle
              r = middle - 1
          elif nums[middle] < target:
              l = middle + 1
          else:
              r = middle - 1
        return first
        
      def findLast(nums, target):
        l, r = 0, len(nums) - 1
        last = -1
        while l <= r:
            middle = (l + r) // 2
            if nums[middle] == target:
                last = middle
                l = middle + 1
            elif nums[middle] < target:
                l = middle + 1
            else:
                r = middle - 1
        return last
      
      first = findfirst(nums, target)
      last = findLast(nums, target)
        
      return [first, last] if first != -1 else [-1, -1]
        